src/pet_mad/explore/_featurizer.py

```diff
@@ -0,0 +1,188 @@
+            from. If provided, the `version` parameter is ignored.
+        :param pet_checkpoint_path: path to a petmad checkpoint file to use for the
+            model from. If not provided, the latest checkpoint is fetched from
+            HuggingFace.
+        :param check_consistency: should we check the model for consistency when
+            running, defaults to False.
+        :param device: torch device to use for the calculation. If `None`, we will try
+            the options in the model's `supported_device` in order.
+        :param length_unit: unit of length used in the structures
+        """
+
+        if version not in VERSIONS:
+            raise ValueError(
+                f"Version {version} is not supported. Supported versions are {VERSIONS}"
+            )
+
+        cache_dir = user_cache_dir("pet-mad", "metatensor")
+        os.makedirs(cache_dir, exist_ok=True)
+
+        petmad_explorer_path = _get_explorer_path(
+            version=version,
+            checkpoint_path=checkpoint_path,
+            cache_dir=cache_dir,
+        )
+
+        petmad = get_pet_mad(version=version, checkpoint_path=pet_checkpoint_path)
+
+        explorer = MADExplorer(petmad, device=device)
+        explorer.load_checkpoint(petmad_explorer_path)
+
+        outputs = {"features": mta.ModelOutput(per_atom=True)}
+        self.dtype = torch.float64
+
+        capabilities = mta.ModelCapabilities(
+            outputs=outputs,
+            length_unit="angstrom",
+            supported_devices=["cpu", "cuda"],
+            dtype=dtype_to_str(self.dtype),
+            interaction_range=0.0,
+            atomic_types=explorer.pet.atomic_types,
+        )
+
+        self.mad_explorer = mta.AtomisticModel(explorer.eval(), METADATA, capabilities)
+
+        self.check_consistency = check_consistency
+        self.device = device
+        self.length_unit = length_unit
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+
+    def __call__(self, frames, environments):
+        systems = mta.systems_to_torch(frames)
+        vesin_metatomic.compute_requested_neighbors(
+            systems,
+            self.length_unit,
+            self.mad_explorer,
+        )
+
+        systems = [s.to(dtype=self.dtype, device=self.device) for s in systems]
+
+        if environments is not None:
+            selected_atoms = mts.Labels(
+                names=["system", "atom"],
+                values=torch.tensor(
+                    [(system, atom) for system, atom, _ in environments]
+                ),
+            )
+        else:
+            selected_atoms = None
+
+        options = mta.ModelEvaluationOptions(
+            length_unit=self.length_unit,
+            outputs={"features": mta.ModelOutput(per_atom=True)},
+            selected_atoms=selected_atoms,
+        )
+
+        outputs = []
+
+        for i_sys in self.progress_bar(range(0, len(systems), self.batch_size)):
```

Comment:
@ceriottm I don't understand why this needs a batch_size. Was it overflowing memory?

Reply:
yes it was, and very quickly - with a 1000-atoms systems, 4 frames would already kill it on CPU on my laptop. I think we do need a batch, and default to 1